lib/active_model/serializer/adapter/cached_serializer.rb

```diff
@@ -27,17 +29,21 @@ def fragment_cached?
           @klass._cache_only && !@klass._cache_except || !@klass._cache_only && @klass._cache_except
         end
 
-        def cache_key
+        def cache_key(adapter_instance)
           parts = []
           parts << object_cache_key
+          parts << adapter_instance.class.to_s.demodulize.underscore
```

Comment:
So, the adapter name is now part of the cache_key. Aside, seems the adapter should define its own name rather than rely on https://github.com/rails-api/active_model_serializers/blob/8aaaa44ba3d5b328fbbf92d59b9f3ecf1b802c8f/lib/active_model/serializer/adapter.rb#L48-L49 making it up...